### PR TITLE
Updated some incomplete docs

### PR DESCRIPTION
**Pull request by mahmudhamid**
#5739  

Updated the following incomplete feature guides:

1. docs/guide/expanding.md
2. docs/guide/fuzzy-filtering.md
3. docs/guide/global-faceting.md
4. docs/guide/global-filtering.md
5. docs/guide/grouping.md

Content is based on my current understanding of the TanStack table library. It may not be full and comprehensive but at least include the essentials.

### CODE REVIEW

**Comment by KevinVandy** [comment]
Impressive following through. Feel free to open up your GitHub sponsors if you live somewhere that allows that, especially if you plan to contribute more.

**Comment by mahmudhamid** [comment]
> Impressive following through. Feel free to open up your GitHub sponsors if you live somewhere that allows that, especially if you plan to contribute more.

Thanks, Kevin. Sponsorships are not available in my region. I'm certainly eager to contribute more whenever possible.

File: docs/guide/expanding.md [hunk 1]
```diff
@@ -13,4 +13,193 @@ Want to skip to the implementation? Check out these examples:
...
+### Expanded rows state
+
+If you need to control the expanded state of the rows in your table, you can do so by using the expanded state and the `onExpandedChange` option. This allows you to manage the expanded state according to your requirements.
+
+```ts
+const [expanded, setExpanded] = useState<ExpandedState>({})
+
+const table = useReactTable({
+  // other options...
+  getState: () => ({
```
Comment (KevinVandy) [inline docs/guide/expanding.md:110]: state instead of getState

File: docs/guide/global-faceting.md [hunk 2]
```diff
@@ -12,4 +12,70 @@ Want to skip to the implementation? Check out these examples:
...
+  getGlobalFacetedRowModel,
+  getGlobalFacetedMinMaxValues, //depends on getGlobalFacetedRowModel
+  getGlobalFacetedUniqueValues, //depends on getGlobalFacetedRowModel
+} from '@tanstack/react-table'
+//...
+const table = useReactTable({
+  // other options...
+  getCoreRowModel: getCoreRowModel(),
+  getGlobalFacetedRowModel: getGlobalFacetedRowModel(), //if you need a list of values for a column (other faceted row models depend on this one)
+  getGlobalFacetedMinMaxValues: getGlobalFacetedMinMaxValues(), //if you need min/max values
```
Comment (KevinVandy) [inline docs/guide/global-faceting.md:36]: These APIs don't actually exist. We only have column specific apis.

File: docs/guide/grouping.md [hunk 3]
```diff
@@ -19,3 +19,127 @@ There are 3 table features that can reorder columns, which happen in the followi
...
+  groupedColumnMode: 'reorder',
+})
+```
+
+### Aggregations
+
+When rows are grouped, you can aggregate the data in the grouped rows by columns using the aggregationFn option. This is a string that is the ID of the aggregation function. You can define the aggregation functions using the aggregationFns option.
+
+```tsx
+const column = columnHelper.data('key', {
```
Comment (KevinVandy) [inline docs/guide/grouping.md:77]: I don't think this is real code either

File: docs/guide/grouping.md [hunk 4]
```diff
@@ -19,3 +19,127 @@ There are 3 table features that can reorder columns, which happen in the followi
...
+### Grouping Change Handler
+
+If you want to manage the grouping state yourself, you can use the onGroupingChange option. This option is a function that is called when the grouping state changes. You can pass the managed state back to the table via the tableOptions.state.grouping option.
+
+```tsx
+const [grouping, setGrouping] = useState<string[]>([])
+
+const table = useReactTable({
+  // other options...
+  getState: () => ({
```
Comment (KevinVandy) [inline docs/guide/grouping.md:140]: state instead of getState

**Comment by KevinVandy** [comment]
It's ok to use AI to help write the docs, but we need to make sure that we don't leave any hallucinations in it.

File: docs/guide/grouping.md (hunk 3 above)
Comment (mahmudhamid) [reply docs/guide/grouping.md]: If you mean the createColumnHelper does not have a data api which I just checked and it doesn't. I copied that piece of code the grouping api doc.

**Comment by mahmudhamid** [comment]
> It's ok to use AI to help write the docs, but we need to make sure that we don't leave any hallucinations in it.

Thank you for the review. I use AI for reviewing English language and it appears there was a confusion between the table.options and the table APIs, leading to some auto-corrections in the code. My Bad.

**Review by KevinVandy — APPROVED** [review]

**Review by Scutua — APPROVED** [review]

**Review by Scutua — APPROVED** [review]

**Pull request merged**